topicgenerator/src/main/java/io/dockstore/topicgenerator/client/cli/TopicGeneratorClient.java

```diff
@@ -194,8 +198,15 @@ private void generateTopics(TopicGeneratorConfig topicGeneratorConfig, GenerateT
                             + " in one sentence that starts with a present tense verb in the <summary> tags. Use a maximum of 150 characters.\n<content>"
                             + descriptorFile.getContent() + "</content>";
                     AIResponseInfo aiResponseInfo = aiModel.get().submitPrompt(prompt);
-                    CSVHelper.writeRecord(csvPrinter, trsId, versionId, descriptorFile, aiResponseInfo);
-                    LOG.info("Generated topic for entry with TRS ID {} and version {}", trsId, versionId);
+                    boolean isCensoredTopic = isSuspiciousTopic(aiResponseInfo.aiResponse());
+                    if (isCensoredTopic) {
+                        // Write censored topics to a different file
+                        CSVHelper.writeRecord(filteredTopicsCsvPrinter, trsId, versionId, descriptorFile, aiResponseInfo);
+                        numberOfCensoredTopics += 1;
+                    } else {
+                        CSVHelper.writeRecord(unfilteredTopicsCsvPrinter, trsId, versionId, descriptorFile, aiResponseInfo);
+                    }
+                    LOG.info("Generated topic for entry with TRS ID {} and version {}.{}", trsId, versionId, isCensoredTopic ? "The topic was filtered because it is potentially offensive" : "");
```

Comment:
You could log the topic here as well... If, for example, there's just one and it's blatantly obvious that it should or should not be filtered, you would just see it in CloudWatch, and not need to download the file. OTOH, we'd be filling our log with potential profanities, lol.

I was thinking it might sometimes save a step of having to find/download the s3 file, but I could be wrong.

Reply:
You would still need to S3 URI to provide to the program so that you can review and approve it, so it may not do much to have the topics in the logs.

If it helps, if you're already looking at the logs, the last few lines of the log stream should have the S3 URI of interest if there are filtered topics:

> Completed 3.2 KiB/4.9 KiB (47.4 KiB/s) with 1 file(s) remaining
Completed 4.9 KiB/4.9 KiB (65.2 KiB/s) with 1 file(s) remaining
upload: ./filtered-topics_CLAUDE_3_HAIKU_20241104T215210Z.csv to s3://qa-dockstore-ai-topics/2024-11-04/f880629f8a114b67bca222ff257c9ba7/filtered-topics_CLAUDE_3_HAIKU_20241104T215210Z.csv